app/src/main/java/edu/mills/heartoftaiwanese/activity/adapter/WordListAdapter.kt

```diff
@@ -0,0 +1,125 @@
+    var favoriteButtonListener: FavoriteButtonListener? = null
+
+    interface FavoriteButtonListener {
+        /**
+         * Interface function that determines what can happen if the favorite is clicked.
+         * The view adapter takes care of hiding/showing the favorite button.
+         *
+         * @return the new value of the favorite
+         */
+        fun onFavoriteClicked(word: DatabaseWord): Boolean
+    }
+
+    inner class WordListViewHolder(private val binding: LayoutWordItemBinding) :
+        RecyclerView.ViewHolder(binding.root) {
+
+        private val dateFormat = DateFormat.getDateInstance(DateFormat.LONG)
+
+        private var isFavorite: Boolean = false
+            set(value) {
+                field = value
+                if (value) {
+                    // Display the favorite button as colorPrimary
+                    binding.favoriteSelector.imageTintList = ColorStateList.valueOf(
+                        binding.root.resources.getColor(R.color.colorPrimaryDark, null)
+                    )
+                } else {
+                    // Unfavorite; clear the favorite button.
+                    binding.favoriteSelector.imageTintList = ColorStateList.valueOf(
+                        binding.root.resources.getColor(R.color.buttonUnselectedGray, null)
+                    )
+                }
+            }
+
+        fun bind(databaseWord: DatabaseWord) {
+            displayWord(databaseWord.word)
+            isFavorite = databaseWord.favorite
+            binding.tvLastAccessed.text = dateFormat.format(databaseWord.accessTime)
+            binding.favoriteSelector.setOnClickListener {
+                favoriteButtonListener?.let {
+                    isFavorite = it.onFavoriteClicked(databaseWord)
+                    notifyDataSetChanged()
+                }
+            }
+        }
+
+        private fun displayWord(word: Word) {
+            binding.tvRightText.text = word.taiwanese
+            binding.tvWordTitle.text = word.taiwanese
+
+            binding.tvCenterText.text = word.chinese
+            binding.tvLeftText.text = word.english
+        }
+    }
+
+    /**
+     * Called when RecyclerView needs a new [ViewHolder] of the given type to represent
+     * an item.
+     *
+     *
+     * This new ViewHolder should be constructed with a new View that can represent the items
+     * of the given type. You can either create a new View manually or inflate it from an XML
+     * layout file.
+     *
+     *
+     * The new ViewHolder will be used to display items of the adapter using
+     * [.onBindViewHolder]. Since it will be re-used to display
+     * different items in the data set, it is a good idea to cache references to sub views of
+     * the View to avoid unnecessary [View.findViewById] calls.
+     *
+     * @param parent The ViewGroup into which the new View will be added after it is bound to
+     * an adapter position.
+     * @param viewType The view type of the new View.
```

Comment:
kdoc descriptions generally don't start with capital letters (unless Intuit uses another convention). https://kotlinlang.org/docs/kotlin-doc.html#module-and-package-documentation

Reply:
This javadoc matches the javadoc of the super class function. It was automatically generated.